src/custom-elements/src/components/number-input/number-input.tsx

```diff
@@ -0,0 +1,182 @@
+    let stepDifference = 0;
+
+    this.value = !!ev.target ? (ev.target as HTMLInputElement).value.toString() : null;
+
+    stepDifference = Math.round((+this.value % this.step - this.initialValue) * 10000) / 10000;
+
+    if (stepDifference !== 0) {
+      this.value = (Math.round(((+this.value - stepDifference) * 10000) / 10000) + this.step).toString();
+    }
+
+    if ((+this.value > this.max) || (+this.value < this.min)) {
+      stepDifference = Math.round((+this.value % this.step - this.initialValue) * 10000) / 10000;
+
+      if (stepDifference !== 0) {
+        this.value = (Math.round(((+this.value - stepDifference) * 10000) / 10000) + this.step).toString();
+      }
+    }
+    
+    this._didUpdateCallBackOnceQueue.push(()=>{
+      this.chiChange.emit(this.value);
+    });
+  }
+
+  private increment() {
+    const step = this.step;
+    const newValue = Math.round((+this.value + step) * 10000) / 10000;
+
+    if (newValue > this.max) {
+      return;
+    } else {
+      this.value =  newValue.toString();
+      this._didUpdateCallBackOnceQueue.push(()=>{
+        this.chiChange.emit(this.value);
+      });
+    }
+  }
+
+  private decrement() {
+    const step = this.step;
+    const newValue = Math.round((+this.value - step) * 10000) / 10000;
+
+    if (newValue < this.min) {
+      return;
+    } else {
+      this.value = newValue.toString();
+      this._didUpdateCallBackOnceQueue.push(()=>{
+        this.chiChange.emit(this.value);
+      });
+    }
+  }
+
+  render() {
+    const base = (
+      <div class="a-inputWrapper">
+        <input type="number"
+               class={`a-input ${this.size ? `-${this.size}` : ''} ${this.inputstyle ? `-${this.inputstyle}` : ''} ${this.state ? `-${this.state}` : ''}`}
+               disabled={this.disabled} 
+               value={this.value}
+               step={this.step}
+               max={this.max}
+               min={this.min}
+               onChange={(ev) => this.handleChange(ev)}/>
+        <button disabled={+this.value - this.step < this.min}
+                onClick={() => this.decrement()}>
+          <div class={`arrow down ${this.size ? `-${this.size}` : ''}`}></div>
+        </button>
+        <button disabled={+this.value + this.step > this.max}
+                onClick={() => this.increment()}>
+          <div class={`arrow up ${this.size ? `-${this.size}` : ''}`}></div>
+        </button>
+      </div>
+    );
+
+    const expanded = (
+      <div class={`m-inputNumber ${this.size ? `-${this.size}` : ''} ${this.pill ? '-pill' : ''}`}>
+        <input type="number"
+               class={`a-input ${this.size ? `-${this.size}` : ''} ${this.inputstyle ? `-${this.inputstyle}` : ''} ${this.state ? `-${this.state}` : ''}`}
+               disabled={this.disabled} 
+               value={this.value}
+               step={this.step}
+               max={this.max}
+               min={this.min}
+               onChange={(ev) => this.handleChange(ev)}/>
```

Comment:
we can store input as a const and use it in base as well as expanded it can reduce our lines of code.